pyvista/core/datasetattributes.py

```diff
@@ -355,5 +917,168 @@ def _raise_index_out_of_bounds(self, index: Any):
                 raise KeyError(f'Array index ({index}) out of range [0, {max_index}]')
 
     def _raise_field_data_no_scalars_vectors(self):
+        """Raise a TypeError if FieldData."""
         if self.association == FieldAssociation.NONE:
-            raise TypeError('vtkFieldData does not have active scalars or vectors.')
+            raise TypeError('FieldData does not have active scalars or vectors.')
+
+    @property
+    def active_scalars_name(self) -> Optional[str]:
+        """Name of the active scalars.
+
+        Examples
+        --------
+        Add two arrays to the mesh point data.
+
+        >>> import pyvista
+        >>> mesh = pyvista.Sphere()
+        >>> mesh.point_arrays['my_data'] = range(mesh.n_points)
+        >>> mesh.point_arrays['my_other_data'] = range(mesh.n_points)
+        >>> mesh.point_arrays.active_scalars_name
+        'my_other_data'
+
+        Set the name of the active scalars.
+
+        >>> mesh.point_arrays.active_scalars_name = 'my_data'
+        >>> mesh.point_arrays.active_scalars_name
+        'my_data'
+
+        """
+        try:
+            return self.GetScalars().GetName()
+        except:
+            return None
+
+    @active_scalars_name.setter
+    def active_scalars_name(self, name: str) -> None:
+        self._raise_field_data_no_scalars_vectors()
+        dtype = self[name].dtype
+        # only vtkDataArray subclasses can be set as active attributes
+        if np.issubdtype(dtype, np.number) or dtype == bool:
+            self.SetActiveScalars(name)
+
+    @property
+    def active_vectors_name(self) -> Optional[str]:
+        """Name of the active scalars.
+
+        Examples
+        --------
+        >>> import pyvista
+        >>> import numpy as np
+        >>> mesh = pyvista.Sphere()
+        >>> mesh.point_arrays['my_data'] = np.random.random((mesh.n_points, 3))
+        >>> mesh.point_arrays.active_scalars_name
+        'my_data'
+
+        """
+        try:
+            return self.GetVectors().GetName()
+        except:
+            return None
+
+    @active_vectors_name.setter
+    def active_vectors_name(self, name: str) -> None:
+        self._raise_field_data_no_scalars_vectors()
+        if name not in self:
+            raise KeyError(f'DataSetAttribute does not contain "{name}"')
+        # verify that the array has the correct number of components
+        n_comp = self.GetArray(name).GetNumberOfComponents()
+        if n_comp != 3:
+            raise ValueError(f'{name} needs 3 components, has ({n_comp})')
+        self.SetActiveVectors(name)
+
+    def __eq__(self, other: Any) -> bool:
+        """Test dict-like equivalency."""
+        # here we check if other is the same class or a subclass of self.
+        if not isinstance(other, type(self)):
+            return False
+
+        if set(self.keys()) != set(other.keys()):
+            return False
```

Comment:
This should probably only be here if int indices are gone for good. Otherwise arguably order matters as well.

Reply:
Keys will always be `str`, so I think this is good.